### ISSUE TITLE
Loading and setting array from server breaks computed accessors

### ISSUE THREAD

**Issue by snowl** [mobxjs/mobx#890]
Browser: Google Chrome 56
Node: 6.10.0
Mobx: 3.1.5

I'm having trouble with MobX where I load data in from an ajax request and then whenever I try to access functions or accessors that are computed, they have been deleted. I assume this is because the MobX "magic" isn't being applied when I set the data.

I've provided some annotated example code below.

**The store**: http://pastebin.com/WSB1khse
**React component**: http://pastebin.com/uvvwEutm
**Example JSON data**
`    {
        "examples": [
        {
            "a": "hello",
            "b": " world"
        }]
    }`

(Sorry for the Pastebin links, I can't put the code in here for some reason).

This will just print out a empty div, instead of the expected "hello world". Is there any way to work around this?

Thanks,
David.

**Comment by urugator** [comment]
Probably same as [#195](https://github.com/mobxjs/mobx-react/issues/195)

**Comment by snowl** [comment]
Changing the call from componentWillMount to componentDidMount doesn't seem to fix this issue.

**Comment by snowl** [comment]
This seems to be a bug with typescript and not MobX.

**Comment by mweststrate** [comment]
Not sure if I follow, how is this related to typescript?

Op do 16 mrt. 2017 00:49 schreef David Diaz <notifications@github.com>:

**Comment by snowl** [comment]
Hi mweststrate,

It seems like when you cast using Typescript it doesn't actually cast (since Typescript doesn't enforce types at runtime, only at compile time). So any prototype properties don't get applied at runtime.

See here: http://stackoverflow.com/questions/22875636/how-do-i-cast-a-json-object-to-a-typescript-class

Sorry for making an issue too early... I closed it once I found my issue. This is not related to MobX at all.

Thanks,
David.
